org.emoflon.ibex.tgg.core.runtime/src/org/emoflon/ibex/tgg/operational/strategies/cc/HandleDependencies.java

```diff
@@ -25,35 +24,42 @@
 	 * key: bundle;
 	 * value: bundles that provide context for this bundle
 	 */
-	THashMap<Integer, ArrayList<Integer>> directDependencies = new THashMap<Integer, ArrayList<Integer>>();
+	Int2ObjectOpenHashMap<ArrayList<Integer>> directDependencies = new Int2ObjectOpenHashMap<ArrayList<Integer>>();
 	
 	/**
 	 * Collection of all cycles detected in dependencies between the bundles;
 	 * key: cycle id; 
 	 * value: list of bundles that cause cycle
 	 */
-	THashMap<Integer, ArrayList<Integer>> cyclicDependencies = new THashMap<Integer, ArrayList<Integer>>();
+	Int2ObjectOpenHashMap<ArrayList<Integer>> cyclicDependencies = new Int2ObjectOpenHashMap<ArrayList<Integer>>();
 	private int cyclicDependencyId = 1;
 	
-	TCustomHashMap<RuntimeEdge, TIntHashSet> edgeToMarkingMatches = new TCustomHashMap<>(
+	Object2ObjectOpenCustomHashMap<RuntimeEdge, IntOpenHashSet> edgeToMarkingMatches = new Object2ObjectOpenCustomHashMap<>(
 			new RuntimeEdgeHashingStrategy());
-	THashMap<EObject, TIntHashSet> nodeToMarkingMatches = new THashMap<>();
-	HashSet<Bundle> appliedBundles;
-	TIntObjectMap<THashSet<EObject>> matchToContextNodes = new TIntObjectHashMap<>();
-	TIntObjectMap<TCustomHashSet<RuntimeEdge>> matchToContextEdges = new TIntObjectHashMap<>();
+	Object2ObjectOpenHashMap<EObject, IntOpenHashSet> nodeToMarkingMatches = new Object2ObjectOpenHashMap<>();
+	ObjectLinkedOpenHashSet<Bundle> appliedBundles;
+	Int2ObjectOpenHashMap<ObjectOpenHashSet<EObject>> matchToContextNodes = new Int2ObjectOpenHashMap<ObjectOpenHashSet<EObject>>();
+	Int2ObjectOpenHashMap<ObjectOpenCustomHashSet<RuntimeEdge>> matchToContextEdges = new Int2ObjectOpenHashMap<ObjectOpenCustomHashSet<RuntimeEdge>>();
+	Int2ObjectOpenHashMap<Bundle> matchToBundle = new Int2ObjectOpenHashMap<Bundle>();
 	
-	public HandleDependencies(HashSet<Bundle> appliedBundles, TCustomHashMap<RuntimeEdge, TIntHashSet> edgeToMarkingMatches, THashMap<EObject, TIntHashSet> nodeToMarkingMatches,
-			TIntObjectMap<THashSet<EObject>> matchToContextNodes,TIntObjectMap<TCustomHashSet<RuntimeEdge>> matchToContextEdges) {
+	public HandleDependencies(ObjectLinkedOpenHashSet<Bundle> appliedBundles, Object2ObjectOpenCustomHashMap<RuntimeEdge, IntOpenHashSet> edgeToMarkingMatches, Object2ObjectOpenHashMap<EObject, IntOpenHashSet> nodeToMarkingMatches,
+			Int2ObjectOpenHashMap<ObjectOpenHashSet<EObject>> matchToContextNodes,Int2ObjectOpenHashMap<ObjectOpenCustomHashSet<RuntimeEdge>> matchToContextEdges) {
 		this.appliedBundles = appliedBundles;
+		appliedBundles
+			.forEach(b -> 
+				b.getAllMatches().stream().forEach(m -> 
+					matchToBundle.put(m,  b)
```

Comment:
I get a warning "The method put(Integer, Bundle) from the type Int2ObjectMap<Bundle> is deprecated". `stream()` is not necessary here, right?

Reply:
I fixed the warning.
But the stream() is necessary unless we change this to using an iterator in a loop...